apps/framework-cli/src/framework/consumption/loader.rs

```diff
@@ -1,50 +1,90 @@
+use super::model::{Consumption, ConsumptionQueryParam, EndpointFile};
+use crate::framework::languages::SupportedLanguages;
+use crate::framework::python::executor::{run_python_program, PythonProgram};
 use crate::project::Project;
-use sha2::{Digest, Sha256};
-use std::{fs, io, path::Path};
-
-use super::model::{Consumption, EndpointFile};
 use crate::utilities::PathExt;
+use serde::Deserialize;
+use sha2::{Digest, Sha256};
+use std::{fs, path::Path};
 
 #[derive(Debug, thiserror::Error)]
 #[non_exhaustive]
 pub enum ConsumptionLoaderError {
     #[error("Failed to open file: {0}")]
-    FailedToOpenFile(#[from] std::io::Error),
+    FailedToOpenFile(std::io::Error),
+    #[error("Failed to load query params: {0}")]
+    FailedToLoadParams(std::io::Error),
 }
 
-pub fn load_consumption(project: &Project) -> Result<Consumption, ConsumptionLoaderError> {
-    let endpoint_files = walkdir::WalkDir::new(project.consumption_dir())
+#[derive(Debug, Deserialize)]
+struct QueryParamOutput {
+    params: Vec<ConsumptionQueryParam>,
+}
+
+pub async fn load_consumption(project: &Project) -> Result<Consumption, ConsumptionLoaderError> {
+    let mut endpoint_files = Vec::new();
+    for f in walkdir::WalkDir::new(project.consumption_dir())
```

Comment:
Nice - are we compatible with nested paths now on the consumption side?

ie if I put something in `apis/test/nested` will the URL reflect that?

Reply:
maybe that was the intent when we implemented it
but i don't think that works in python now because i don't see code that maps the `/` (in path) to `.` (python module)